src/library-authoring/add-content/AddContentContainer.tsx

```diff
@@ -184,35 +183,36 @@ const AddContentContainer = () => {
       showToast(intl.formatMessage(messages.successPasteClipboardMessage));
     }).catch((error) => {
       showToast(parseErrorMsg(
+        intl,
         error,
         messages.errorPasteClipboardMessageWithDetail,
         messages.errorPasteClipboardMessage,
       ));
     });
   };
-
   const onCreateBlock = (blockType: string) => {
-    createBlockMutation.mutateAsync({
-      libraryId,
-      blockType,
-      definitionId: `${uuid4()}`,
-    }).then((data) => {
-      const hasEditor = canEditComponent(data.id);
-      if (hasEditor) {
-        // linkComponent on editor close.
-        openComponentEditor(data.id, () => linkComponent(data.id));
-      } else {
+    const suportedEditorTypes = Object.values(blockTypes);
+    if (suportedEditorTypes.includes(blockType)) {
+      // linkComponent on editor close.
+      openComponentEditor('', (data) => linkComponent(data.id), blockType);
```

Comment:
If I create a new "Text" component and then cancel without saving changes, this is giving an error:

```
Cannot read properties of undefined (reading 'id')
```

Reply:
I made the change to fix this